### PR TITLE
CPP Transformer Tutorial

### PR DESCRIPTION
**Pull request by hlahkar**
This is Transformer example using CPP frontend

### CODE REVIEW

File: advanced_source/transformer__timeseries_cpp_tutorial/transformer_timeseries.cpp [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+    auto model = TransformerModel();
+    model.to(device);
+
+    auto data = get_data();
+    train(model, std::get<0>(data));
+    evaluate(model, std::get<1>(data));
+
+    return 0;
+}
```
Comment (glaringlee) [inline advanced_source/transformer__timeseries_cpp_tutorial/transformer_timeseries.cpp:238]: nit: please add a new line here to avoid lint error

**Review by glaringlee — COMMENTED** [review]
Thanks a lot for contributing this tutorial. I have a nit comment at the end to avoid lint error. Will approve once this is corrected.

Not required for this PR. It would be great if you can write a .rst file to go over this example, since .rst file will be available on https://pytorch.org/tutorials/ as a text tutorial.

for example: https://github.com/pytorch/tutorials/blob/master/advanced_source/rpc_ddp_tutorial.rst

**Comment by hlahkar** [comment]
> Thanks a lot for contributing this tutorial. I have a nit comment at the end to avoid lint error. Will approve once this is corrected. Pleases rebase the code as well.
> 
> Not required for this PR. It would be great if you can write a .rst file to go over this example, since .rst file will be available on https://pytorch.org/tutorials/ as a text tutorial.
> 
> for example: https://github.com/pytorch/tutorials/blob/master/advanced_source/rpc_ddp_tutorial.rst

Sure I will be raising a new PR for the .rst file

File: advanced_source/transformer__timeseries_cpp_tutorial/transformer_timeseries.cpp (hunk 1 above)
Comment (hlahkar) [reply advanced_source/transformer__timeseries_cpp_tutorial/transformer_timeseries.cpp]: Have added the newline at the end

**Review by glaringlee — APPROVED** [review]
LGTM

**Pull request merged**